presto-plugin-toolkit/src/main/java/com/facebook/presto/plugin/base/security/StatsRecordingSystemAccessControl.java

```diff
@@ -0,0 +1,729 @@
+    private final SystemAccessControl delegate;
+
+    public StatsRecordingSystemAccessControl(SystemAccessControlStatsRecorder stats, SystemAccessControl delegate)
+    {
+        this.stats = requireNonNull(stats, "stats is null");
+        this.delegate = requireNonNull(delegate, "delegate is null");
+
+        // Register all methods for stats tracking
+        registerMethods();
+    }
+
+    @VisibleForTesting
+    public SystemAccessControl getDelegate()
+    {
+        return delegate;
+    }
+
+    private void registerMethods()
+    {
+        // Register all methods that we want to track
+        stats.register(delegate, "checkCanSetUser");
+        stats.register(delegate, "selectAuthorizedIdentity");
+        stats.register(delegate, "checkQueryIntegrity");
+        stats.register(delegate, "checkCanSetSystemSessionProperty");
+        stats.register(delegate, "checkCanAccessCatalog");
+        stats.register(delegate, "filterCatalogs");
+        stats.register(delegate, "checkCanCreateSchema");
+        stats.register(delegate, "checkCanDropSchema");
+        stats.register(delegate, "checkCanRenameSchema");
+        stats.register(delegate, "checkCanShowSchemas");
+        stats.register(delegate, "filterSchemas");
+        stats.register(delegate, "checkCanShowCreateTable");
+        stats.register(delegate, "checkCanCreateTable");
+        stats.register(delegate, "checkCanSetTableProperties");
+        stats.register(delegate, "checkCanDropTable");
+        stats.register(delegate, "checkCanRenameTable");
+        stats.register(delegate, "checkCanShowTablesMetadata");
+        stats.register(delegate, "filterTables");
+        stats.register(delegate, "checkCanShowColumnsMetadata");
+        stats.register(delegate, "filterColumns");
+        stats.register(delegate, "checkCanAddColumn");
+        stats.register(delegate, "checkCanDropColumn");
+        stats.register(delegate, "checkCanRenameColumn");
+        stats.register(delegate, "checkCanSelectFromColumns");
+        stats.register(delegate, "checkCanInsertIntoTable");
+        stats.register(delegate, "checkCanDeleteFromTable");
+        stats.register(delegate, "checkCanTruncateTable");
+        stats.register(delegate, "checkCanUpdateTableColumns");
+        stats.register(delegate, "checkCanCreateView");
+        stats.register(delegate, "checkCanRenameView");
+        stats.register(delegate, "checkCanDropView");
+        stats.register(delegate, "checkCanCreateViewWithSelectFromColumns");
+        stats.register(delegate, "checkCanSetCatalogSessionProperty");
+        stats.register(delegate, "checkCanGrantTablePrivilege");
+        stats.register(delegate, "checkCanRevokeTablePrivilege");
+        stats.register(delegate, "checkCanDropConstraint");
+        stats.register(delegate, "checkCanAddConstraint");
+        stats.register(delegate, "getRowFilters");
+        stats.register(delegate, "getColumnMasks");
```

Comment:
I changed my mind. Instead of using this pattern to register method names, can you instead create real getters in `SystemAccessControlStatsRecorder` for each method ? Annotating them with `@Managed` should be enough to achieve their JMX export